sdk/src/fee_calculator.rs

```diff
@@ -46,4 +135,54 @@ mod tests {
         let message = Message::new(vec![ix0, ix1]);
         assert_eq!(FeeCalculator::new(2).calculate_fee(&message), 4);
     }
+
+    #[test]
+    fn test_fee_calculator_derived_default() {
+        solana_logger::setup();
+
+        let mut f0 = FeeCalculator::default();
+        assert_eq!(f0.target_signatures_per_slot, 0);
+        assert_eq!(f0.target_lamports_per_signature, 0);
+        assert_eq!(f0.lamports_per_signature, 0);
+        f0.target_lamports_per_signature = 42;
+
+        let f1 = FeeCalculator::new_derived(&f0, 4242);
+        assert_eq!(f1.target_signatures_per_slot, 0);
+        assert_eq!(f1.target_lamports_per_signature, 42);
+        assert_eq!(f1.lamports_per_signature, 42);
+    }
+
+    #[test]
+    fn test_fee_calculator_derived_adjust() {
```

Comment:
I'm a little surprised not to see any asserts in this test. Would it not be helpful to ensure that the ramps are landing on the desired fee rate, and perhaps use a counter to ensure how many iterations it takes to get there? Or are the limits and the fee adjustment too volatile to bother with this?

Reply:
I wrote this test with `while` loops intended on it simply not terminating if something goes wrong.   I didn't want to make assumptions about the rate of fee change here since I figured we'd probably be tweaking this based on feedback during TdS stages/etc.  But the three loops do ensure the fee gets to the max/min/desired amount.